### DISCUSSION TITLE
HowTo create Patterns with more than 16/4 time?

### DISCUSSION THREAD

**Discussion by gorgonz** [hydrogen-music/hydrogen#2001] [Q&A] [answered]
I try add a song to hydrogen, where some pattern have 20/4 time. Hydrogen will refuse to add bigger values than 16/4 time.
Do I have to "Rebase" the configuration of hydrogen in some way or is it impossible? What is the reason for this limit?
Thx in advance for tips or advices :-)

**Comment by theGreatWhiteShark** [comment, marked as answer]
Hey @gorgonz ,

> What is the reason for this limit?

That's an excellent question! I have no idea :D . This upper limit of four 4/4 blocks does exist since the onset of our git history 17 years ago. Maybe handling longer patterns did not worked with UI design back then.

Anyway. I do not see a reason why we should hold on to this limit. (At times it bothers me as well). But dropping it could have quite some side effects. That's why I won't integrate it in the upcoming 1.2.4 release but in 2.0, which is still way further away.

Till then you could patch the code and compile it yourself. AFAICS the only thing you need to do is to replace `4 *` with a larger number in

https://github.com/hydrogen-music/hydrogen/blob/releases/1.2/src/gui/src/PatternEditor/PatternEditorPanel.cpp#L1022
https://github.com/hydrogen-music/hydrogen/blob/releases/1.2/src/gui/src/PatternEditor/PatternEditorPanel.cpp#L1050
https://github.com/hydrogen-music/hydrogen/blob/releases/1.2/src/gui/src/PatternEditor/PatternEditorRuler.cpp#L60
https://github.com/hydrogen-music/hydrogen/blob/releases/1.2/src/gui/src/PatternEditor/PatternEditorRuler.cpp#L543
https://github.com/hydrogen-music/hydrogen/blob/releases/1.2/src/gui/src/PatternEditor/PatternEditorRuler.cpp#L561

**Comment by gorgonz** [comment]
Hey @theGreatWhiteShark ,

Your solution is a very friendly way to help me in my troubles, thx a lot, I appreciate it :-)

I must wait until the weekend, then I give it a try :-)

In return I will report, whether or not it ended up in problems or other effects ;-)

**Comment by gorgonz** [comment]
Well, You and the community are so much assisting :-). Sauerland added the new build (together with the lib) to his Home Project https://download.opensuse.org/repositories/home:/Sauerland:/mixx/15.6/

I only needed to install from there, here the first results:

- runs like  a charm
- one hint only: the time counter still will end up at 16, but the new time strokes are there and work flawlessly :-)

**Reply by theGreatWhiteShark** [reply]
> Sauerland added the new build (together with the lib) to his Home Project https://download.opensuse.org/repositories/home:/Sauerland:/mixx/15.6/

A build containing the 4 changes I suggested above?

> one hint only: the time counter still will end up at 16, but the new time strokes are there and work flawlessly :-)

Yeah, I also noted that. But fixing this is a little more elaborate than changing a number. I will have a look at it for the v2.0 release.

**Comment by gorgonz** [comment]
Yes this build has the 4 changes with version 1.2.3  (https://build.opensuse.org/projects/home:Sauerland:mixx/packages/hydrogen/files/createnewlimit.patch?expand=1)

I'm totally fine with the counter thing, was just a note ;-). Do You have an estimated time of delivery for 2.0?

**Reply by theGreatWhiteShark** [reply]
> Yes this build has the 4 changes with version 1.2.3 (https://build.opensuse.org/projects/home:Sauerland:mixx/packages/hydrogen/files/createnewlimit.patch?expand=1)

wow

> Do You have an estimated time of delivery for 2.0?

Hmm. Hard to say since bugs have higher priority and some of them require quite some time. I guess the final 2.0.0 release including documentation and web page update might be done somewhere in summer/autumn 2025. But maybe a beta version might be ready till the end of the year? Hard to say. Still too many things on the TODO list.

**Comment by theGreatWhiteShark** [comment]
Implementation will be tracked in #2005
